### ISSUE TITLE
[Bug]: Build error on NetBSD (ldl)

### ISSUE THREAD

**Issue by ktnb-netbsd** [naev/naev#2570]
### What happened?

While working on porting the latest version to NetBSD, I ran across this error during configure.
```
meson.build:196:16: ERROR: C shared or static library 'dl' not found
```
NetBSD doesn't have libdl but sometimes has these issues with cmake and meson. While we have a way to bypass built into pkgsrc it would be great if this could be fixed upstream. :)

### Version

0.11.4

### What operating system are you seeing the problem on?

BSD

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bobbens** [comment]
Does https://github.com/naev/naev/commit/4fcef9bed7ff532cbbfbbf3706d743e337238cca solve it?

**Comment by UncombedCoconut** [comment]
Depending on distro status, we might be able to assume a more recent Meson and use `dependency('dl')` as opposed to `find_library`: https://mesonbuild.com/Dependencies.html#dl-libdl

**Comment by ktnb-netbsd** [comment]
> Thanks for the report. Does [4fcef9b](https://github.com/naev/naev/commit/4fcef9bed7ff532cbbfbbf3706d743e337238cca) solve it?

So long that dlopen isn't actually required for certain parts of the game, it appears that it works!

**Comment by bobbens** [comment]
> Depending on distro status, we might be able to assume a more recent Meson and use `dependency('dl')` as opposed to `find_library`: https://mesonbuild.com/Dependencies.html#dl-libdl

Cool. Adding a comment about that in meson.build

> So long that dlopen isn't actually required for certain parts of the game, it appears that it works!

I'm actually not sure. However, since it seems to work fine, let us close this issue and open a new one as necessary. Thanks again.

**Comment by bobbens** [comment]
We released 0.11.5 that contains the fix.

**Comment by ktnb-netbsd** [comment]
> We released 0.11.5 that contains the fix.

Thank you, we are not up-to-date: https://mail-index.netbsd.org/pkgsrc-changes/2024/06/14/msg301843.html
